datafusion/functions/src/unicode/substr.rs

```diff
@@ -180,32 +197,26 @@ fn string_view_substr(
 
     let start_array = as_int64_array(&args[0])?;
 
+    // In either case of `substr(s, i)` or `substr(s, i, cnt)`
+    // If any of input argument is `NULL`, the result is `NULL`
     match args.len() {
         1 => {
-            for (idx, (raw, start)) in string_view_array
-                .views()
+            for ((str_opt, raw_view), start_opt) in string_view_array
```

Comment:
The old implementation will treat empty string in view  as `NULL` (within `make_and_append_view()`), so it handled `NULL` case correctly  but handled the empty string case wrong.
Since it's not possible to use view to check if the current element is null, so here we also have to iterate on the string array to check whether each element is `NULL`

Reply:
Another potential thing that might be faster is to iterate on the null array directly (rather than the `Option<&str>`) so the code doesn't have to figure out where the `&str` comes from

Reply:
Thanks for correcting my mistakes. I really appreciate it! I'll be more careful with my code next time.

Reply:
> Another potential thing that might be faster is to iterate on the null array directly (rather than the `Option<&str>`) so the code doesn't have to figure out where the `&str` comes from

Great point, there are also other potential optimizations like specializing not null columns, scalar literal arguments, and ASCII case. So I prefer only to make a simple and correct implementation in this PR, and evaluate those optimizations as a future task